File: PSP/springjavafx/src/main/java/org/example/springjavafx/DIJavafx.java
package org.example.springjavafx;


import javafx.application.Application;
import javafx.application.Platform;
import javafx.stage.Stage;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.context.ApplicationEvent;
import org.springframework.context.ConfigurableApplicationContext;


public class DIJavafx extends Application {

    private ConfigurableApplicationContext applicationContext;

    @Override
    public void init() throws Exception {
        super.init();
        applicationContext = new SpringApplicationBuilder(SpringjavafxApplication.class).run();
    }

    @Override
    public void stop() throws Exception {
        super.stop();
        applicationContext.close();
        Platform.exit();
    }

    @Override
    public void start(Stage primaryStage) throws Exception {

        primaryStage.setMinWidth(800);
        primaryStage.setMinHeight(600);
        primaryStage.setResizable(true);
        applicationContext.publishEvent(new StageReadyEvent(primaryStage));

    }


    static class StageReadyEvent extends ApplicationEvent {
        public StageReadyEvent(Stage stage) {
            super(stage);
        }

        public Stage getStage() {
            return ((Stage) getSource());
        }
    }
}

File: PSP/graphql/frontend/src/main/kotlin/org/example/frontend/DiJavaFX.kt
package org.example.frontend

import javafx.application.Application
import javafx.application.Platform
import javafx.stage.Stage
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.ApplicationEvent
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.stereotype.Component

@Component
class DiJavaFX (

) : Application(){

    private lateinit var applicationContext: ConfigurableApplicationContext

    override fun init() {
        super.init()
        applicationContext = SpringApplicationBuilder(FrontendApplication::class.java).run()
    }


    override fun stop() {
        super.stop()
        applicationContext.close()
        Platform.exit()
    }


    override fun start(primaryStage: Stage) {
        primaryStage.minWidth = 800.0
        primaryStage.minHeight = 600.0
        primaryStage.isResizable = true
        applicationContext.publishEvent(StageReadyEvent(primaryStage))
    }


    class StageReadyEvent(stage: Stage) : ApplicationEvent(stage) {
        val stage: Stage
            get() = (getSource() as Stage)
    }
}
